src/module.c

```diff
@@ -3719,6 +3719,14 @@ ValkeyModuleString *VM_GetClientUserNameById(ValkeyModuleCtx *ctx, uint64_t id)
     return str;
 }
 
+/* Returns 1 if commands are arriving from the primary client or AOF client
+ * and should never be rejected.
+ * Returns 0 otherwise (and also if ctx or if the client is NULL). */
+int VM_MustObeyClient(ValkeyModuleCtx *ctx) {
+    if (!ctx || !ctx->client) return 0;
```

Comment:
My only fear is that it will become confusing since many times the context holds a fake/tmp client which will not be obeyed right? I think this is why in most cases were module API was checking the mustObeyClient it was targeting the server.current_client. Not sure this is a big concern but maybe we should allow somehow to also check the current running client? like  `int VM_MustObeyCurrentRunningClient()`?

Reply:
Yeah, I looked through the code and didn't see any specific examples where we would need to find the real client. Replication from the primary should not block or rely on background behavior.